include/bluetooth/buf.h

```diff
@@ -51,10 +52,29 @@ struct bt_buf_data {
 #define BT_BUF_RESERVE CONFIG_BT_HCI_RESERVE
 #endif
 
+/** Helper to include reserved HCI data in buffer calculations */
 #define BT_BUF_SIZE(size) (BT_BUF_RESERVE + (size))
 
-/** Data size neeed for HCI RX buffers */
-#define BT_BUF_RX_SIZE (BT_BUF_SIZE(CONFIG_BT_RX_BUF_LEN))
+/** Helper to calculate needed buffer size for HCI ACL packets */
+#define BT_BUF_ACL_SIZE(size) BT_BUF_SIZE(BT_HCI_ACL_HDR_SIZE + (size))
+
+/** Helper to calculate needed buffer size for HCI Event packets. */
+#define BT_BUF_EVT_SIZE(size) BT_BUF_SIZE(BT_HCI_EVT_HDR_SIZE + (size))
+
+/** Helper to calculate needed buffer size for HCI Command packets. */
+#define BT_BUF_CMD_SIZE(size) BT_BUF_SIZE(BT_HCI_CMD_HDR_SIZE + (size))
+
+/** Data size needed for HCI ACL RX buffers */
+#define BT_BUF_ACL_RX_SIZE BT_BUF_ACL_SIZE(CONFIG_BT_BUF_ACL_RX_SIZE)
+
+/** Data size needed for HCI Event RX buffers */
+#define BT_BUF_EVT_RX_SIZE BT_BUF_EVT_SIZE(CONFIG_BT_BUF_EVT_RX_SIZE)
```

Comment:
Yeah, could be confusing, but I don't agree that adding HCI clears it up.

The RX/TX defines are the calculated values, the ones without are the helpers.

Reply:
@joerchan Are the helpers supposed to be used by the applications or other modules? 

I'm just wondering if instead of declaring these as `#define`s in a public header, they should have been functions such that we can hide whatever the application doesn't need (or rather, what we don't want the application) to use.

Reply:
These have to be defines, as they're used in the definition of various buffer arrays. Most (if not all) of these go into buffers that need to be allocated by the application or other high level users, so I they'll need to be in header under include, even though they don't all strictly have to be this front and center. 

I suggest we leave this as-is, any transfer to a separate header would just be cosmetic, and can be done later